src/serial.rs

```diff
@@ -78,40 +102,65 @@ pub enum Error {
     _Extensible,
 }
 
-
-// USART REMAPPING, see: https://www.st.com/content/ccc/resource/technical/document/reference_manual/59/b9/ba/7f/11/af/43/d5/CD00171190.pdf/files/CD00171190.pdf/jcr:content/translations/en.CD00171190.pdf
-// Section 9.3.8
-pub trait Pins<USART> {
-    const REMAP: u8;
+mod sealed {
+    pub trait Remap {
+        type Periph;
+        const REMAP: u8;
+    }
+    pub trait TxPin<REMAP> {}
+    pub trait RxPin<REMAP> {}
+    pub struct _Tx;
+    pub struct _Rx;
```

Comment:
This could use some documentation. Suggestion:

```rust
    /// Implemented for a struct which defines a possible remaping
    /// of the `Periph` peripheral
    pub trait Remap {
        type Periph;
        const REMAP: u8;
    }
    /// Implemented for the gpio pins that can be used as TX with a specific
    /// Remaping
    pub trait TxPin<REMAP> {}
    pub trait RxPin<REMAP> {}
    /// NOTE: Not sure what these do, care to explain?
    pub struct _Tx;
    pub struct _Rx;
```